pkg/sql/distsql_running.go

```diff
@@ -168,26 +214,20 @@ func (dsp *DistSQLPlanner) setupFlows(
 		// into the local flow.
 	}
 	if firstErr != nil {
-		if _, ok := errors.If(firstErr, func(err error) (v interface{}, ok bool) {
-			v, ok = err.(*distsqlrun.VectorizedSetupError)
-			return v, ok
-		}); ok && evalCtx.SessionData.Vectorize == sessiondata.VectorizeOn {
+		if shouldFallbackOnVectorizedSetupError(evalCtx.SessionData.Vectorize, firstErr) {
 			// Error vectorizing remote flows, try again with off.
 			// Generate a new flow ID so that any remote nodes that successfully set
 			// up a vectorized flow will not be connected to by a non-vectorized flow.
-			newFlowID := uuid.MakeV4()
+			oldFlowID := flows[thisNodeID].FlowID
+			newFlowID, reset := prepareVectorizedFlowsForReplanning(evalCtx, oldFlowID, flows)
+			defer reset()
 			log.Infof(
-				ctx, "error vectorizing remote flow %s, restarting with vectorize=off and ID %s: %+v", flows[thisNodeID].FlowID, newFlowID, firstErr,
+				ctx, "error vectorizing remote flow %s, restarting with vectorize=off and ID %s: %+v", oldFlowID, newFlowID, firstErr,
 			)
```

Comment:
This ends up producing a ton of logs because the error has an embedded stack trace. I think we should consider making this hidden behind a vmodule, or at least not using +v on the error output.

For example:

```
I190718 22:43:42.554117 304 sql/distsql_running.go:252  [n1,client=[::1]:58098,user=root] error vectorizing gateway flow 00000000-0000-0000-0000-000000000000, restarting with vectorize=off and ID 00000000-0000-0000-0000-000000000000: error while setting up columnar execution
  - error with attached stack trace:
    github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).setupVectorized
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/column_exec_setup.go:1166
    github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).setup
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:504
    github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ServerImpl).setupFlow
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/server.go:463
    github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ServerImpl).SetupLocalSyncFlow
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/server.go:519
    github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).setupFlows
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:245
    github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:347
    github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:941
    github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:902
    github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:734
    github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:417
    github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:99
    github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1204
    github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1140
    github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:442
    github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:580
    runtime.goexit
        /usr/local/Cellar/go/1.12.4/libexec/src/runtime/asm_amd64.s:1337
  - error with embedded safe details: unable to vectorize execution plan.
  - unable to vectorize execution plan.:
  - error with attached stack trace:
    github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.newColOperator
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/column_exec_setup.go:499
    github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).setupVectorized
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/column_exec_setup.go:1164
    github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).setup
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:504
    github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ServerImpl).setupFlow
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/server.go:463
    github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ServerImpl).SetupLocalSyncFlow
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/server.go:519
    github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).setupFlows
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:245
    github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:347
    github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:941
    github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:902
    github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:734
    github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:417
    github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:99
    github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1204
    github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1140
    github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:442
    github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1
        /Users/jordan/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:580
    runtime.goexit
        /usr/local/Cellar/go/1.12.4/libexec/src/runtime/asm_amd64.s:1337
  - error with embedded safe details: unsupported processor core %s
    -- arg 1: <*distsqlpb.ProcessorCoreUnion>
  - unsupported processor core localPlanNode:<RowSourceIdx:0 NumInputs:0 Name:"virtual table" >
```